### PR TITLE
ARROW-5442: [Website] Clarify what makes a release artifact "official"

### PR DESCRIPTION
**Pull request by nealrichardson**

### CODE REVIEW

File: site/install.md [hunk 1]
```diff
@@ -37,39 +37,40 @@ See the [release notes][10] for more about what's new.
...
 
-### C++ and Python Conda Packages (Unofficial)
+We have provided official binary wheels on PyPI for Linux, macOS, and Windows:
+
+```shell
+pip install pyarrow=={{site.data.versions['current'].pinned_number}}
+```
+
+We recommend pinning `{{site.data.versions['current'].pinned_number}}`
+in `requirements.txt` to install the latest patch release.
```
Comment (nealrichardson) [inline site/install.md]: Can we remove this? Perhaps an artifact of a bygone unstable time? I'd think we'd want to discourage version pinning (people who want/need to pin will do so anyway).

File: site/install.md [hunk 2]
```diff
@@ -210,7 +211,7 @@ sudo yum install -y --enablerepo=epel parquet-devel # For Apache Parquet C++
 sudo yum install -y --enablerepo=epel parquet-glib-devel # For Parquet GLib (C)
 ```
 
-### C++ and GLib (C) Packages for MSYS2 (Unofficial)
+### C++ and GLib (C) Packages for MSYS2
 
 The official MSYS2 packages include [Apache Arrow C++ and GLib (C)
```
Comment (nealrichardson) [inline site/install.md]: I'll note that this seems in conflict: heading said "unofficial" but the sentence says "official"

File: site/install.md [hunk 3]
```diff
@@ -84,7 +85,7 @@ and GLib (C) package with:
 brew install apache-arrow-glib
 ```
 
-### C++ and GLib (C) Packages for Debian GNU/Linux, Ubuntu and CentOS (Unofficial)
+### C++ and GLib (C) Packages for Debian GNU/Linux, Ubuntu and CentOS
```
Comment (kou) [inline site/install.md]: They are "unofficial" just for 0.13.0 by packaging mistake.
https://lists.apache.org/thread.html/43ecf510ea3f04a375854f70886195c140d0e96854b43286e389e3c0@%3Cdev.arrow.apache.org%3E
Normally, this is "official" (voted) binary packages.

File: site/install.md (hunk 2 above)
Comment (kou) [reply site/install.md]: I meant this is "official MSYS2 packages" and isn't "official Apache Arrow packages".
"official MSYS2 packages" means that they are included in https://github.com/msys2/MINGW-packages .

Anyway, we can remove the "official".

File: site/install.md (hunk 1 above)
Comment (kou) [reply site/install.md]: We can remove this when we reach 1.0.0.
API isn't stable until 1.0.0.

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comments?

File: site/install.md (hunk 3 above)
Comment (nealrichardson) [reply site/install.md]: Ok, so we can move this above the "unofficial" marker when we do 0.14.0.

**Comment by nealrichardson** [comment]
Pushed, PTAL.

File: site/install.md (hunk 3 above)
Comment (nealrichardson) [reply site/install.md]: https://issues.apache.org/jira/browse/ARROW-5444 for that fix later.

File: site/install.md (hunk 1 above)
Comment (nealrichardson) [reply site/install.md]: https://issues.apache.org/jira/browse/ARROW-5445

I'm noticing that for every issue I pick up, I end up making two or three more 🤔

**Review by kou — APPROVED** [review]
+1

File: site/install.md (hunk 1 above)
Comment (wesm) [reply site/install.md]: Version pinning is commonplace in the Python world -- I don't think API stability has much to do with it (we will still have some API changes or deprecations after 1.0 I would guess)

Comment (kou) [reply site/install.md]: OK. I've closed https://issues.apache.org/jira/browse/ARROW-5445 .

**Pull request closed without merging**